src/supremm/supremmconf.py

```diff
@@ -1,31 +1,43 @@
 #!/usr/bin/env python
 """ helper utiility to print out config info """
+from __future__ import print_function
 
 import sys
 import os
 import json
+import logging
 from getopt import getopt
 
 from supremm.config import Config
+from supremm.scripthelpers import setuplogger
 
 def usage():
     """ print usage """
-    print "usage: {0} [OPTS]".format(os.path.basename(__file__))
-    print "  -s --section SECTION  output the configuration data from the specified section"
-    print "  -i --item ITEM        output the configuration data for the specified item"
-    print "  -h --help             print this help message"
-
+    print("usage: {0} [OPTS]".format(os.path.basename(__file__)))
+    print("  -d --debug            set log level to debug")
+    print("  -c --config           specify the configuration file")
+    print("  -r --resources        output the configuration data for all enabled resources") 
+    print("  -s --section SECTION  output the configuration data from the specified section")
+    print("  -i --item ITEM        output the configuration data for the specified item")
+    print("  -h --help             print this help message")
 
 def getoptions():
     """ process comandline options """
 
-    retdata = {"section": None, "item": None}
+    retdata = {"log"	: logging.ERROR,
+               "config"	: None,
+               "section": None,
+               "item"	: None}
 
-    opts, _ = getopt(sys.argv[1:], "s:i:h", ["section=", "item="])
+    opts, _ = getopt(sys.argv[1:], "dc:s:i:h", ["debug", "config=", "section=", "item="])
 
     for opt in opts:
+        if opt[0] in ("-d", "--debug"):
+            retdata['log'] = logging.DEBUG 
+        if opt[0] in ("-c", "--config"):
+            retdata['config'] = opt[1]
         if opt[0] in ("-s", "--section"):
-            retdata['section'] = opt[1].encode("utf-8")
+            retdata['section'] = opt[1]
```

Comment:
How come the `encode("utf-8")` function has been removed?

Reply:
When I was working with the python3 version this script wasn't able to read my configuration unless that was removed. I carried over all the changes from the py3 updates to this PR, so it might not be necessary for this version of the software.

Reply:
Just added that back in

Reply:
Just thought I'd mention that the utf-8 encoding was removed in the py3 version because of this error. Top screenshot is with encoding and bottom screenshot is without encoding.
![image](https://user-images.githubusercontent.com/51850219/222834299-5f6724e8-9719-43ab-9ac1-307bd5354779.png)
![image](https://user-images.githubusercontent.com/51850219/222834344-24442d45-8a6b-438c-9e4a-37755da9c52e.png)
The code works with or without the utf-8 encoding for py2 so my thought process was to just remove it altogether.